### PR TITLE
add role-grantor access to cluster-bot service account

### PR DESCRIPTION
**Pull request by hoxhaeris**
Grant Cluster-bot roles to allow giving access to cluster initiators.

2 new roles for the cluster-bot service account:

1. Grant permission that allows creating RoleBinding objects

```ymal
- apiGroups:
  - rbac.authorization.k8s.io
  resources:
  - rolebindings
  verbs: :
  - create
 ```

2. Grant permission (explicitly) by giving permission to bind that particular ClusterRole

```yaml
- apiGroups:
  - rbac.authorization.k8s.io
  resources:
  - clusterroles
  verbs:
  - bind
  resourceNames:
  - admin
```

### CODE REVIEW

File: clusters/build-clusters/common/ci-chat-bot/ci-chat-bot.yaml [hunk 1]
```diff
@@ -44,3 +44,34 @@ subjects:
...
+  resources:
+  - rolebindings
+  verbs:
+  - create
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterroles
+  verbs:
+  - create
```
Comment (bradmwilliams) [inline clusters/build-clusters/common/ci-chat-bot/ci-chat-bot.yaml:64]: These can be collapsed down into a single group with multiple `resources`

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

**Review by bradmwilliams — APPROVED** [review]

**Comment by hongkailiu** [comment]
/hold

Could you explain why the permissions are needed, especially of app.ci?

**Comment by hongkailiu** [comment]
Could the whole file be moved to https://github.com/openshift/release/tree/master/clusters/build-clusters/common_except_app.ci?

Which cluster roles will be created by the bot?
Can they be created as a static yaml in release repo?

**Comment by hoxhaeris** [comment]
@hongkailiu 

>Could you explain why the permissions are needed, especially of app.ci?

Cluster-Bot will grant access to the cluster initiators (on the respective temporary namespace).
The bot will create a role+rolebinding on runtime (on the temporary namespace), to grant access to the cluster initiator. 

>Which cluster roles will be created by the bot?

The cluster roles will be bound to the temporary namespace only. 

>Can they be created as a static yaml in release repo?

No, we can't do that, since the roles will be created on the temporary namespaces.

**Comment by hoxhaeris** [comment]
@hongkailiu 

> Could the whole file be moved to https://github.com/openshift/release/tree/master/clusters/build-clusters/common_except_app.ci?

We can't move the whole file, but we can move the new ClusterRole to `common_except_app.ci`

**Comment by hongkailiu** [comment]
> The cluster roles will be bound to the temporary namespace only.

It explains the `rolebindings`. I cannot see why it needs the permission on creating clusterroles.

**Comment by hoxhaeris** [comment]
> > The cluster roles will be bound to the temporary namespace only.
> 
> It explains the `rolebindings`. I cannot see why it needs the permission on creating clusterroles. Maybe an example of the clusterrole that might be created?

We will create a ClusterRole specifically for this purpose: to allow access to the cluster initiators on the temporary namespace. 
The ClusterRole will contain something like: 
```golang
	role := &rbacapi.ClusterRole{
		ObjectMeta: m,
		Rules: []rbacapi.PolicyRule{
			{
				APIGroups: []string{""},
				Resources: []string{"pods/attach", "pods/exec", "pods/portforward", "pods/proxy", "services/proxy"},
				Verbs:     []string{"get", "list", "watch"},
			},
		},
	}
```

**Comment by hoxhaeris** [comment]
@hongkailiu 
We made a few changes to the PR, by removing the need to create the ClusterRoles.

The current change creates 2 new ClusterRoles:
1. Grants permission that allows creating RoleBinding objects (this part has not changed)
2. Explicitly grant permission to the service account to bind the `admin` resourceName in a given namespace. 

During runtime, we will bind the admin ClusterRole to the cluster-initiator user on the temporary namespace.

**Comment by hongkailiu** [comment]
@hoxhaeris 

Feel free to cancel the hold whenever ready.

**Pull request merged**